### PR TITLE
[SPARK-8792] [ML] Add Python API for PCA transformer

### PR DESCRIPTION
**Pull request by yanboliang**
Add Python API for PCA transformer

### CODE REVIEW

**Comment by yanboliang** [comment]
@mengxr @jkbradley

**Comment by yanboliang** [comment]
@mengxr @jkbradley

**Comment by jkbradley** [comment]
I'll make a pass now

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -1030,6 +1030,67 @@ class Word2VecModel(JavaModel):
...
+        Set params for this PCA.
+        """
+        kwargs = self.setParams._input_kwargs
+        return self._set(**kwargs)
+
+    def setK(self, value):
+        """
+        Sets the value of :py:attr:`k`.
+        """
+        self._paramMap[self.k] = value
```
Comment (jkbradley) [inline python/pyspark/ml/feature.py]: return self

**Comment by jkbradley** [comment]
@yanboliang It looks great, save for that one issue.

**Comment by jkbradley** [comment]
LGTM, merging with master.  Thank you!

Could you please make & link a JIRA for adding a Python example to the mllib-feature-extraction.md user guide section?

**Comment by yanboliang** [comment]
@jkbradley I have open #7522 to add user guide and code example for ml.PCA. I think we should put them in ml-features.md. Actually the Scala/Java examples are also missing and I have add them as well.

**Pull request closed without merging**